### DISCUSSION TITLE
leaderboard score_val

### DISCUSSION THREAD

**Discussion by talhaanwarch** [autogluon/autogluon#1170] [Q&A] [answered]
Greeings, @Innixma It is  not clear to me what is meant score test and score val in leader board.
I have trained a model as follow
```
clf=TabularPredictor(label='Label',verbosity=0).fit(train,tuning_data=val)
res=clf.leaderboard(test,silent=False)
```
In leaderboard, i just sent a test data, so I am expecting a single score column . But I am confuse from where the val score comes from?

**Comment by Innixma** [comment, marked as answer]
`score_val` is the validation score on the `tuning_data`. It is AutoGluon's best guess at how strong a model is prior to being given test data. You would see this same column if you did `clf.leaderboard()` without specifying test data. AutoGluon picks the model with the best `score_val` is the `best_model` which it uses in `.predict` and `.predict_proba` calls by default.

**Reply by Innixma** [reply]
Note that saying `score_val` is the score on the `tuning_data` is a bit a simplification, as when stack ensembling is enabled, `score_val` is actually the cross-validation score on `train_data` derived from out-of-fold predictions.

For more information, refer to the AutoGluon-Tabular paper: https://arxiv.org/pdf/2003.06505.pdf

**Reply by talhaanwarch** [reply]
so if i do `clf.leaderboard(val,silent=False)` then test_score and val_score should be the same, provided that no stack ensembling take place

**Reply by Innixma** [reply]
That is correct

**Reply by MaximAliev** [reply]
@Innixma what could be the reason that the dataframe's score_val column contains Nones?
